src/common/lib/client/auth.ts

```diff
@@ -374,67 +383,87 @@ class Auth {
    * - requestHeaders (optional, unsupported, for testing only) extra headers to add to the
    *                  requestToken request
    */
-  async requestToken(tokenParams: API.TokenParams | null, authOptions: API.AuthOptions): Promise<API.TokenDetails>;
+  async requestToken(tokenParams: API.TokenParams | null, authOptions: AuthOptions): Promise<API.TokenDetails>;
 
-  async requestToken(tokenParams?: API.TokenParams | null, authOptions?: any): Promise<API.TokenDetails> {
+  async requestToken(tokenParams?: API.TokenParams | null, authOptions?: AuthOptions): Promise<API.TokenDetails> {
     /* RSA8e: if authOptions passed in, they're used instead of stored, don't merge them */
-    authOptions = authOptions || this.authOptions;
-    tokenParams = tokenParams || Utils.copy(this.tokenParams);
+    const resolvedAuthOptions = authOptions || this.authOptions;
+    const resolvedTokenParams = tokenParams || Utils.copy(this.tokenParams);
 
     /* first set up whatever callback will be used to get signed
      * token requests */
-    let tokenRequestCallback: any,
+    let tokenRequestCallback: (
+        data: API.TokenParams,
+        callback: (
+          error: API.ErrorInfo | RequestResultError | string | null,
```

Comment:
This is because `tokenRequestCallback` here needs to accept `AuthOptions.authCallback`, which allows you to pass a string error to the callback:

 https://github.com/ably/ably-js/blob/7f4e17cff564497ec8c29715efebf303c0c128a6/ably.d.ts#L593

(_Why_ it allows that, I don’t know, but seems outside scope of this PR.)

Reply:
Yeah, I've seen `AuthOptions.authCallback` too, wondered if you know the reason for that.
Seems like a legacy thing, not going to change it here